warehouse/packaging/services.py

```diff
@@ -583,15 +579,19 @@ def create_project(
             )
 
         # Remove all pending publishers not owned by the creator.
-        # There might be other pending publishers for the same project name,
-        # which we've now invalidated by creating the project. These would
+        # There might be other pending publishers for the same or similar project
+        # name, which we've now invalidated by creating the project. These would
         # be disposed of on use, but we explicitly dispose of them here while
         # also sending emails to their owners.
         stale_pending_publishers = (
             request.db.query(PendingOIDCPublisher)
             .filter(
-                func.normalize_pep426_name(PendingOIDCPublisher.project_name)
-                == func.normalize_pep426_name(project.name),
+                or_(
+                    func.normalize_pep426_name(PendingOIDCPublisher.project_name)
+                    == func.normalize_pep426_name(project.name),
+                    func.ultranormalize_name(PendingOIDCPublisher.project_name)
+                    == func.ultranormalize_name(project.name),
+                ),
```

Comment:
I think the 'ultranormalized' set will include all of the 'normalized' matches as well? This `or_` condition might not be necessary.

Reply:
I thought it safer to match on both since there is no formal relationship between the functions that guarantees that to be the case, but I will drop it per your preference.